### DISCUSSION TITLE
Dirint Application Problem

### DISCUSSION THREAD

**Discussion by thokpp** [pvlib/pvlib-python#1806] [Q&A]
Hello Guys,
I have a problem with the application of dirint which is :
-> Dirint not give me the right 'DNI' when i use it like this
<img width="549" alt="image" src="https://github.com/pvlib/pvlib-python/assets/78738099/1249af55-b923-4af7-b868-bdc0af08a979">

=> dirint = pvlib.irradiance.dirint(ghi, zenith, times, pressure=dataframe['barometric_pressure'],
temp_dew=dataframe['temp_dew'])
(pressure is in Bar and Temperature in Celsius)          
Whereas, erbs model gives me the right value for same data.
I don't understand why. If you can help me it will be nice and i will be very thanksfull.
Thank you for your reading guys =)

PS : Here is my first line of data : 
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/hp6482/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/hp6482/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">

_time | air_temperature | barometric_pressure | Observed POA | irradiance_ghi | relative_humidity | apparent_zenith | zenith | apparent_elevation | elevation | azimuth | equation_of_time
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
2023-05-31 11:57:00+00:00 | 18,7799999 | 992,540002 | 84,54 | 50,8 | 98,8800003 | 84,2035451 | 84,349743 | 5,79645491 | 5,65025704 | 67,6589433 | 2,32497354

</body>

</html>

**Comment by cwhanse** [comment]
I think the barometric_pressure is in units of kPa, not Pa.

**Comment by thokpp** [comment]
Thank you, i have the sensation that's work for DNI but on my second line i would like to calculate DHI and this value is bugging too.
Do you have any idea which can help me ?
Thank you Hanse :)

**Reply by kandersolar** [reply]
Depending on how `zenith` was calculated, that could be a units issue as well; pvlib returns solar position angles in degrees while `np.cos` takes its argument in radians: https://numpy.org/doc/stable/reference/generated/numpy.cos.html

If `zenith` is in degrees, you can use `np.cos(np.radians(zenith))`.  Or use `dhi = pvlib.irradiance.complete_irradiance(zenith, ghi=ghi, dni=dirint)`, which does essentially the same thing.

**Comment by thokpp** [comment]
PS : for sure, i corrected kPa in Pa now =)
